scan.go

```diff
@@ -918,6 +911,32 @@ func stringToCm(token string) (e, m uint8, ok bool) {
 	return
 }
 
+func nameToAbsolute(name, origin string) (absolute string, err string) {
```

Comment:
There is also dnsutil.AddOrigin. Maybe we should just add a Join there which is a much better name.

Reply:
`err` was a string from attempting to bubble up specific errors, but in the end I had to stick with generic messages like "bad origin", so I converted this to a bool. And I don't think I understand the second comment; maybe it can be followup?

Reply:
2nd: maybe we should add a dnsutil.Join that peforms the above (`nameToAbsolute`) contatenation?

Reply:
this would also fix the `err string` thing as we would just need to return an actual error.

Reply:
https://github.com/miekg/dns/issues/525